crates/bevy_utils/src/parallel_queue.rs

```diff
@@ -39,21 +39,24 @@ where
     I: IntoIterator<Item = T> + Default + Send + 'static,
 {
     /// Collect all enqueued items from all threads and them into one
-    pub fn drain<B>(&mut self) -> B
+    /// 
+    /// The ordering is not guarenteed.
+    pub fn drain<B>(&mut self) -> impl Iterator<Item = T> + '_
```

Comment:
I think it's worth just documenting the behavior, since I think it'll be more common that users will just drain the entire thing without prematurely terminating the iteration.